heat/core/manipulations.py

```diff
@@ -1686,11 +1686,15 @@ def reshape(a, shape, new_split=None):
     (1/2) tensor([[0., 2., 4., 6.]])
     (2/2) tensor([[ 8., 10., 12., 14.]])
     """
-    if not isinstance(a, dndarray.DNDarray):
-        raise TypeError("'a' must be a DNDarray, currently {}".format(type(a)))
-    if not isinstance(shape, (list, tuple)):
-        raise TypeError("shape must be list, tuple, currently {}".format(type(shape)))
-        # check new_split parameter
+    sanitation.sanitize_in(a)
+    # check for nested sequence
+    if isinstance(shape[0], (list, tuple)):
+        shape = shape[0]
+    else:
+        if not isinstance(shape, (list, tuple)):
+            raise TypeError("shape must be list, tuple, currently {}".format(type(shape)))
```

Comment:
Can we get an exception test for this?

Reply:
@Markus-Goetz , I salute you. Indeed, this exception is not being raised. Any idea why?